File: app/src/main/java/com/paradigmadigital/dagger/di/DetailActivityComponent.java
package com.paradigmadigital.dagger.di;

import com.paradigmadigital.dagger.platform.ActivityModule;
import com.paradigmadigital.dagger.platform.DetailActivityModule;
import com.paradigmadigital.dagger.ui.detail.DetailActivity;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@PerActivity
@Subcomponent(modules = { ActivityModule.class, DetailActivityModule.class})
public interface DetailActivityComponent extends AndroidInjector<DetailActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<DetailActivity> {
    }
}

File: module_detail/src/main/java/eu/sesma/dagger/detail/DetailActivityComponent.kt
package eu.sesma.dagger.detail

import android.support.v7.app.AppCompatActivity
import dagger.Component
import eu.sesma.dagger.core.ActivityScoped
import eu.sesma.dagger.core.CoreComponent


@ActivityScoped
@Component(
        dependencies = [CoreComponent::class],
        modules = [DetailActivityModule::class]
)
interface DetailActivityComponent {

    fun inject(activity: DetailActivity)

    //Exposed to sub-graphs.
    fun activity(): AppCompatActivity
}